aws: check for all possible Terminated states

tomwans opened the pull request:
**What type of PR is this?**
/kind bug

**What this PR does / why we need it:**
It's possible that an EC2 instance is in the Terminated state for multiple autoscaler loops, so to avoid trying to terminate it more than once and causing the bug seen for this initial fix, we include this check as well.

Also, make the log more verbose and write out what state we found the instance in.

This builds upon this change: https://github.com/kubernetes/autoscaler/pull/5411

drmorr0 commented on cluster-autoscaler/cloudprovider/aws/aws_cloud_provider_test.go:
```diff
@@ -504,6 +504,54 @@ func TestDeleteNodesTerminatingInstances(t *testing.T) {
+
+	newSize, err := asgs[0].TargetSize()
+	assert.NoError(t, err)
+	assert.Equal(t, 2, newSize)
```
Why isn't this `expectedInstancesCount`?

tomwans replied:
Going to double check, but I just copy pasted the tests from the last PR for this and added the new Terminated state; will double check both.

tomwans commented:
@drmorr0 thanks for the review. I cleaned up the test a bit, please let me know how it looks now. It just seems like the TerminatingWait test was copy pasted from something else, with light adjustments. The expectedInstanceCount variable is not really relevant, it was just a distraction. I reworded some of that from the Terminated test I added.

If you'd like, I can copy paste this and do the same for the TerminatingWait test, so they look the same. Or, I can just bring them all in to one test and do a table test. I have no preference. What do you think?

drmorr0 commented:
> @drmorr0 thanks for the review. I cleaned up the test a bit, please let me know how it looks now. It just seems like the TerminatingWait test was copy pasted from something else, with light adjustments. The expectedInstanceCount variable is not really relevant, it was just a distraction. I reworded some of that from the Terminated test I added.
> 
> If you'd like, I can copy paste this and do the same for the TerminatingWait test, so they look the same. Or, I can just bring them all in to one test and do a table test, which might be fine since we expect the same result out of all the states anyway. I have no preference. What do you think?

@tomwans If you have some time and it's not terribly difficult, putting into a table test would be nice, I guess?  But I don't have particularly strong feelings on the matter, so if you don't have time/want to just ship it, I'll stamp it.

tomwans commented:
Let's just stamp it out. Theres no other case of table tests in this test file, so it wouldn't be matching conventions anyway.

drmorr0 commented:
/lgtm
/approve
